plugin/audit_log/audit_log.c

```diff
@@ -1446,6 +1448,14 @@ static struct st_mysql_sys_var* audit_log_system_variables[] =
   NULL
 };
 
+char thd_local_init_buf[sizeof(audit_log_thd_local)];
```

Comment:
THDVAR_SET makes a copy of provided string and stores it in thread variables. So, there are two options:
1. Create static buffer, fill it with sizeof(audit_log_local) bytes, pass it to THDVAR_SET each time
2. Allocate dynamic buffer and fill it for every new THD, pass it to THDVAR_SET and then release it.
I went with first option.